chapters/data-processing.tex

```diff
@@ -0,0 +1,784 @@
+about the parts involved in each transaction on the same table.
+The next two sections will outline steps to check if your data is tidy,
+and to tidy it if it is not.
+
+\subsection{Tidying raw data}
+
+% Is the uniquely identified data tidy?
+Sometimes, making sure that data is uniquely and fully identified 
+is enough to yield a tidy dataset.
+This happens when a data source or survey instrument
+contains a single unit of observation.
+If you are working with data in this format,
+you may skip the tidying process,
+which can become quite intricate.
+In this case the cleaning process will result in a single dataset.
+
+% Examples of untidy raw data
+The most common case of untidy raw data encountered in development research
+are datasets consisting of multiple units of observation stored in the same table. 
+If your rows include multiple nested observational units,
+having an identifying variable for each row doesn't mean having an identifier for each observation.
+This is the case of our previous example of untidy data:
+a survey that collects both household and household-member level information.
+In this case, 
+the easiest way to handle the data is often to break it into two tidy datasets:
+one at the household level, and one at the member level.
+You will use these separate datasets for the rest of the data cleaning process.
+
+% How to tidy data
+Creating these two tidy datasets will involve a few different transformations.
+Data collection platforms such as SurveyCTO usually export \textbf{wide datasets},
+where each row represents the highest level of observation,
+and variables observed in nested units are separated into one column per sub-unit,
+such as \texttt{age\_1}, \texttt{gender\_1}, \texttt{age\_2}, \texttt{gender\_2}, etc.
+To create tidy datasets from this data,
+you first need to identify which variables were observed at which levels.
+In our example, the presence of a refrigerator was collected at household level (the higher unit),
+while age and gender where observed at the member sub-unit.
+These two sets of variables will be separated.
+Create a tidy dataset containing only the unique row ID and the household level variables.
+Then create another dataset containing only the unique row ID and the household-member level variables.
+Note that the household-member dataset is still not tidy.
+It needs to be \textbf{reshaped}\sidenote{\textbf{Reshape:}
+	transform a dataset in such a way that the unit of observation represented by a row changes.}
+to a \textbf{long} format.
+The reshaped data set will have a new variable indicating the member ID,
+and its unique identifier will be the combination of household ID and member ID.
+This is a tidy dataset, where
+the \texttt{gender} and \texttt{age} variables are each represented by one column.
+The reshape can be done using \texttt{reshape} in Stata, and \texttt{reshape2} or \texttt{pivot\_longer} in R.
```

Comment:
it would be nice if we could point to a code example (but i don't know if there's any example that would be easily digestible by an outsider)

Reply:
moved to #466